src/coprocessor/xeval/builtin_cast.rs

```diff
@@ -0,0 +1,398 @@
+            invalid_type_error};
+
+fn check_datum_int(datum: &Datum) -> Result<()> {
+    match *datum {
+        Datum::I64(_) | Datum::U64(_) => Ok(()),
+        _ => Err(Error::Eval(format!("invalid expr type: {:?}, expect to be int", datum))),
+    }
+}
+
+impl Evaluator {
+    pub fn cast_int_as_int(&mut self, ctx: &EvalContext, expr: &Expr) -> Result<Datum> {
+        let child = try!(self.get_one_child(expr));
+        let d = try!(self.eval(ctx, child));
+        match d {
+            Datum::I64(_) => {
+                if has_unsigned_flag(expr.get_field_type().get_flag() as u64) {
+                    Ok(Datum::U64(d.u64()))
+                } else {
+                    Ok(d)
+                }
+            }
+            Datum::U64(u) => {
+                if has_unsigned_flag(expr.get_field_type().get_flag() as u64) {
+                    Ok(d)
+                } else {
+                    let i = try!(convert::convert_uint_to_int(u, i64::MAX));
+                    Ok(Datum::I64(i))
+                }
+            }
+            _ => invalid_type_error(&d, TYPE_INT),
+        }
+    }
+
+    pub fn cast_int_as_real(&mut self, ctx: &EvalContext, expr: &Expr) -> Result<Datum> {
+        let child = try!(self.get_one_child(expr));
+        let d = try!(self.eval(ctx, child));
+        match d {
+            Datum::I64(i) => Ok(Datum::F64(i as f64)),
+            Datum::U64(u) => Ok(Datum::F64(u as f64)),
+            _ => invalid_type_error(&d, TYPE_INT),
+        }
+    }
+
+    pub fn cast_int_as_string(&mut self, ctx: &EvalContext, expr: &Expr) -> Result<Datum> {
+        let child = try!(self.get_one_child(expr));
+        let d = try!(self.eval(ctx, child));
+        match d {
+            Datum::I64(_) | Datum::U64(_) => {}
+            _ => return invalid_type_error(&d, TYPE_INT),
+        }
+        let s = try!(d.into_string());
+        let s = try!(produce_str_with_specified_tp(s, expr.get_field_type(), ctx));
+        Ok(Datum::Bytes(s.into_bytes()))
+    }
+
+    pub fn cast_int_as_decimal(&mut self, ctx: &EvalContext, expr: &Expr) -> Result<Datum> {
+        let child = try!(self.get_one_child(expr));
+        let datum = try!(self.eval(ctx, child));
+        let decimal = match datum {
+            Datum::I64(_) => Decimal::from(datum.i64()),
+            Datum::U64(_) => Decimal::from(datum.u64()),
+            _ => return invalid_type_error(&datum, TYPE_INT),
+        };
+        let decimal = try!(produce_dec_with_specified_tp(decimal, expr.get_field_type(), ctx));
+        Ok(Datum::Dec(decimal))
+    }
+
+    pub fn cast_int_as_time(&mut self, ctx: &EvalContext, expr: &Expr) -> Result<Datum> {
+        let child = try!(self.get_one_child(expr));
+        let d = try!(self.eval(ctx, child));
+        try!(check_datum_int(&d));
+        let s = try!(d.into_string());
+        let fsp = expr.get_field_type().get_decimal() as i8;
+        let mut t = try!(Time::parse_datetime(&s, fsp, &ctx.tz));
```

Comment:
It seems [`parse_datetime`](https://github.com/pingcap/tikv/blob/master/src/coprocessor/codec/mysql/time.rs#L178) is the same with tidb's [`parseDatetime`](https://github.com/pingcap/tidb/blob/master/util/types/time.go#L535), and should we wrap a function named [`parseTime`](https://github.com/pingcap/tidb/blob/master/util/types/time.go#L1135) as `tidb` did?